### PR TITLE
Minor refactoring of chaos_test

### PR DESCRIPTION
**Pull request by vivekmenezes**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4057)

### CODE REVIEW

File: acceptance/chaos_test.go [hunk 1]
```diff
@@ -141,20 +151,20 @@ func TestChaos(t *testing.T) {
 					break
 				default:
 				}
-				if time.Now().After(deadline) {
+				if done() {
```
Comment (vivekmenezes) [inline acceptance/chaos_test.go]: minor change in functionality, but I don't see why not.

File: acceptance/chaos_test.go [hunk 2]
```diff
@@ -75,31 +83,36 @@ func TestChaos(t *testing.T) {
...
-					err := pErr.GoError()
-					if _, ok := err.(*roachpb.SendError); ok || testutils.IsError(err, rpc.ErrShutdown.Error()) || testutils.IsError(err, "client is unhealthy") {
-						// Common errors we can ignore. Also suppress the
-						// log messages so they don't get spammy.
-						curErrorOutput := fmt.Sprintf("client %d: %s", i, err)
-						if prevErrorOutput != curErrorOutput {
-							log.Warning(curErrorOutput)
-							prevErrorOutput = curErrorOutput
+			for !done() {
+				if err := func() error {
```
Comment (tschottdorf) [inline acceptance/chaos_test.go]: I like avoiding large unnamed closures. Could you pull this out into a function?

Comment (vivekmenezes) [inline acceptance/chaos_test.go]: I could but it's going to involve passing in a few parameters like k, value, r, which makes the function awkward.

**Comment by tschottdorf** [comment]
LGTM mod that locking comment.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[acceptance/chaos_test.go, line 87 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4057#-K9SVN9DtcJXq9gxjQc6-r1-87:-K9T-Mhc87FDne_2uuOr:1937616000) ([raw file](https://github.com/cockroachdb/cockroach/blob/5485e63bb89b2f01079124776ce6c0528185a173/acceptance/chaos_test.go#L87)):*
ok, I don't feel too strongly.

---

*[acceptance/chaos_test.go, line 133 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4057#-K9T-69ell7nn7kji5_B:-K9T-69f1-rELk4sHvTq:668045347) ([raw file](https://github.com/cockroachdb/cockroach/blob/c1588e14b9dcf9164e1d3f7e5f5e55f1be47001b/acceptance/chaos_test.go#L133)):*
why this change? Before, we halted all clients to avoid killing nodes while clients were writing (since that could let them get stuck ). Maybe we've since fixed those issues, but I'd hate to have them come back here.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4057#-:-K9T-R5s8AZTLyM0D8Pn:-581358100)*

**Comment by vivekmenezes** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/chaos_test.go, line 133 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4057#-K9T-69ell7nn7kji5_B:-K9T0fFIxCsGZ_V3UHje:852885504) ([raw file](https://github.com/cockroachdb/cockroach/blob/c1588e14b9dcf9164e1d3f7e5f5e55f1be47001b/acceptance/chaos_test.go#L133)):*
I see. I thought we only needed to stop the clients that were talking directly to the nodes being restarted, but I see how once the node acts as a proxy for a request it might get stuck. One could argue that we would uncover a bug in that case right.

I'll revert this change, but this entire test leaves a bad taste in my mouth. I feel that we should probably change it to kill a fewer nodes, and in the process check that clients are able to write at all times (with very few errors). Freezing client activity is just wrong.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4057)*

**Comment by tschottdorf** [comment]
> I'll revert this change, but this entire test leaves a bad taste in my mouth.

I agree, but the our decided fix for a lot of the bad error handling in the face of connectivity issues is that the move to `gRPC` will address many of them, so it hasn't made sense to address this so far.

**Pull request merged**